sql/structured.proto

```diff
@@ -73,6 +73,13 @@ message IndexDescriptor {
   // parallels the column_names list.
   repeated uint32 column_ids = 5 [(gogoproto.customname) = "ColumnIDs",
       (gogoproto.casttype) = "ColumnID"];
+  // An ordered list of additional column ids to use for the index. For
+  // non-unique indexes, these columns will be appended to the key. For unique
+  // indexes these columns will be stored in the value. The extra column IDs is
+  // computed as PrimaryIndex.column_ids - column_ids. For the primary index
+  // the list will be empty.
+  repeated uint32 extra_column_ids = 6 [(gogoproto.customname) = "ExtraColumnIDs",
```

Comment:
> DisambiguatingColumnIDs is as verbose as PrimaryMinusColumnIDs. Do we suddenly start using a language other than Go?

I'm OK with any other name that conveys what this field is for - I don't feel as strongly as you on the subject of verbosity.

> What is the harm in storing this on the descriptor? Storing it here is effectively acting as a persistent cache. If we didn't store it here we'd have a more complex interface to encodeSecondaryIndexes if we wanted to avoid recomputing this list on every row inserted/deleted/updated.

I'm fine with this, given that the primary key can't be changed.